src/ProjectOrigin.Vault/Repositories/CertificateRepository.cs

```diff
@@ -162,6 +162,7 @@ CREATE TEMPORARY TABLE certificates_work_table ON COMMIT DROP AS (
                     AND (@start IS NULL OR start_date >= @start)
                     AND (@end IS NULL OR end_date <= @end)
                     AND (@type IS NULL OR certificate_type = @type)
+                    AND quantity != 0
```

Comment:
No. But this is because a certificate can end up with 0 quantity if the slices expire. You can see the certificates_query_model view in one of the sql files